ClientApp/viking-vault/src/components/AccountPage.tsx

```diff
@@ -51,41 +78,101 @@ class AccountPage extends React.Component<any, IAccountState>{
                     if(response.status === 404){
                         this.setState(oldState => {
                             return {
-                                isPresent: false
+                                isPresent: true
                             }
                         })
                     }
                     if(response.status === 200){
                         this.setState(oldState => {
                             return {
-                                isPresent: true
+                                isPresent: false
                             }
                         })
                     }
-                    response.json();})
-                .then(data=>{})
+                    return response.json();})
+                .then((userData:any)=>{
+                    console.log(userData)
+                    if(userData != null){
+                        this.setState(
+                            {
+                                firstName: userData.firstName,
+                                lastName: userData.lastName,
+                                cardNumber: userData.cardNumber,
+                                expirationDate: userData.expirationDate,
+                                ronBalance: userData.ronBalance,
+                                eurBalance: userData.eurBalance,
+                                usdBalance: userData.usdBalance,
+                                yenBalance: userData.yenBalance
+                            }
+                        )
+                    }
+                })
                 .catch(error => {
                     this.setState(oldState => {
```

Comment:
Same here, you don't need to call `setState` with the `oldState` object if you don't use it